datasource-sdk/src/main/java/org/apache/kylin/sdk/datasource/adaptor/PrestoAdaptor.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kylin.sdk.datasource.adaptor;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Types;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.sql.rowset.CachedRowSet;
+
+public class PrestoAdaptor extends DefaultAdaptor {
+
+    private Pattern patternASYM = Pattern.compile("BETWEEN(\\s*)ASYMMETRIC");
+    private Pattern patternSYM = Pattern.compile("BETWEEN(\\s*)SYMMETRIC");
+    private Pattern patternTrim = Pattern.compile("TRIM\\(.*BOTH.*FROM\\s+(.+)\\)");
+    private Pattern patternOffset = Pattern.compile("(?i)OFFSET\\s\\d+");
+
+    public PrestoAdaptor(AdaptorConfig config) throws Exception {
+        super(config);
+    }
+
+    @Override
+    public String fixSql(String sql) {
+        sql = resolveBetweenAsymmetricSymmetric(sql);
+        sql = convertTrim(sql);
+        sql = convertOffset(sql);
+        return sql;
+    }
+
+    @Override
+    public int toKylinTypeId(String type, int typeId) {
+        if (2000 == typeId) {
```

Comment:
Could you please kindly explain the magic number `2000` means? I check the [JDK7 Constant value](https://docs.oracle.com/javase/7/docs/api/constant-values.html), 2000 is refer to `JAVA_OBJECT`, am I right?

Reply:
Looks like it is presto's specific behaviour of mapping `DECIMAL` to `JAVA_OBJECT`.